### PR TITLE
docs: Update releasing.md

### PR DESCRIPTION
**Pull request by TeodorSAP**
## Description

Changes proposed in this pull request (what was done and why):

- Clarify (potentially ambiguous) requirement in the release process documentation.

Changes refer to particular issues, PRs or documents:

- 

## Traceability
- [ ] The PR is linked to a GitHub issue.
- [ ] New features have a milestone set.
- [ ] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [ ] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [ ] Adjusted the documentation if the change is user-facing.
- [ ] The feature is unit-tested
- [ ] The feature is e2e-tested

### CODE REVIEW

**Review by chrkl — COMMENTED** [review]
Please use the `docs` prefix for this kind of PR.

File: docs/contributor/releasing.md [hunk 1]
```diff
@@ -32,16 +32,22 @@ This release process covers the steps to release new major and minor versions fo
 
 6. Merge the PR.
 
-7. In the `telemetry-manager/{RELEASE_BRANCH}` branch, create release tags for the head commit.
+7. Make sure that the release tags will point to the HEAD commit of the `telemetry-manager/{RELEASE_BRANCH}` branch. This can be done, by rebasing the upstream branch into the local branch after the merge was successful.
```
Comment (NHingerl) [inline docs/contributor/releasing.md:35]:
```suggestion
7. To make sure that the release tags point to the HEAD commit of the `telemetry-manager/{RELEASE_BRANCH}` branch, rebase the upstream branch into the local branch after the merge was successful.
```

Do we really need "after the merge was successful"? Merging is the previous step; would anyone really do step 7 before they see the merge has happened successfully?

File: docs/contributor/releasing.md [hunk 2]
```diff
@@ -32,16 +32,22 @@ This release process covers the steps to release new major and minor versions fo
...
+
+   ```bash
+   git rebase upstream/{RELEASE_BRANCH} {RELEASE_BRANCH}
+   ```
+
+8. In the `telemetry-manager/{RELEASE_BRANCH}` branch, create release tags for the head commit.
```
Comment (NHingerl) [inline docs/contributor/releasing.md:41]:
```suggestion
8. In the `telemetry-manager/{RELEASE_BRANCH}` branch, create release tags for the HEAD commit.
```

This was in all caps before, so let's stay consistent.

File: docs/contributor/releasing.md (hunk 1 above)
Comment (TeodorSAP) [reply docs/contributor/releasing.md:35]: I did it yesterday 🥲

**Comment by TeodorSAP** [comment]
I will be closing this PR, since I'm having issues changing the file. Please refer to this one instead: https://github.com/kyma-project/telemetry-manager/pull/1053

**Pull request closed without merging**